nilearn/decomposition/multi_pca.py

```diff
@@ -1,115 +1,47 @@
+from sklearn.base import TransformerMixin
 
-from ..input_data import NiftiMasker, MultiNiftiMasker, NiftiMapsMasker
-from ..input_data.nifti_masker import filter_and_mask
-from .._utils.class_inspect import get_params
-from .._utils.cache_mixin import cache, CacheMixin
-from .._utils import as_ndarray
-from .._utils.compat import _basestring
-
-
-def session_pca(imgs, mask_img, parameters,
-                n_components=20,
-                confounds=None,
-                memory_level=0,
-                memory=Memory(cachedir=None),
-                verbose=0,
-                copy=True):
-    """Filter, mask and compute PCA on Niimg-like objects
-
-    This is an helper function whose first call `base_masker.filter_and_mask`
-    and then apply a PCA to reduce the number of time series.
-
-    Parameters
-    ----------
-    imgs: list of Niimg-like objects
-        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-        List of subject data
-
-    mask_img: Niimg-like object
-        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-        Mask to apply on the data
-
-    parameters: dictionary
-        Dictionary of parameters passed to `filter_and_mask`. Please see the
-        documentation of the `NiftiMasker` for more informations.
-
-    confounds: CSV file path or 2D matrix
-        This parameter is passed to signal.clean. Please see the
-        corresponding documentation for details.
-
-    n_components: integer, optional
-        Number of components to be extracted by the PCA
-
-    memory_level: integer, optional
-        Integer indicating the level of memorization. The higher, the more
-        function calls are cached.
-
-    memory: joblib.Memory
-        Used to cache the function calls.
-
-    verbose: integer, optional
-        Indicate the level of verbosity (0 means no messages).
-
-    copy: boolean, optional
-        Whether or not data should be copied
-    """
-
-    data, affine = cache(
-        filter_and_mask, memory,
-        func_memory_level=2, memory_level=memory_level,
-        ignore=['verbose', 'memory', 'memory_level', 'copy'])(
-            imgs, mask_img, parameters,
-            memory_level=memory_level,
-            memory=memory,
-            verbose=verbose,
-            confounds=confounds,
-            copy=copy)
-    if n_components <= data.shape[0] // 4:
-        U, S, _ = randomized_svd(data.T, n_components)
-    else:
-        U, S, _ = linalg.svd(data.T, full_matrices=False)
-    U = U.T[:n_components].copy()
-    S = S[:n_components]
-    return U, S
+from .._utils.cache_mixin import CacheMixin
+from .base import BaseDecomposition, mask_and_reduce
 
 
-class MultiPCA(BaseEstimator, TransformerMixin, CacheMixin):
+class MultiPCA(BaseDecomposition, TransformerMixin, CacheMixin):
     """Perform Multi Subject Principal Component Analysis.
+    This is a good initialization method for ICA.
```

Comment:
Newline needed here, see [PEP 0257](https://www.python.org/dev/peps/pep-0257/#id17). "Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description"